### PR TITLE
WIP: Looking For Group system update

### PR DESCRIPTION
**Pull request by Zerixx**
As the title says this is a work in progress. 

Goals: 
- Add, update, and implement packet structures.
- Update old methods and remove deprecated and or un-useful code.
- Add new technology to the LFG/DF system. E.g., Premade Finder.
- Make sure it works properly.
- More to be added as we get to it, or it's thought up.

Do not merge the pull request just yet.

I would LOVE it if other people with the similar interest in this system would like to contribute to the progress. It's a shame to see a great system no longer supported and updated for the current patch, or even expansion, at this point. 

If you notice any mistakes, errors, or issues, I would appreciate it if you could point them out to me, and if possible a solution. And remember, this is a WIP, and in NO MEANS is it functional, or working as intended. 

-Z

### CODE REVIEW

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 1]
```diff
@@ -36,3 +36,354 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
 
     return data;
 }
+
+ByteBuffer& operator>>(ByteBuffer& data, WorldPackets::LFG::LfgBootInfo& bootInfo)
+{
+    data >> bootInfo.VoteInProgress;
```
Comment (Runico) [inline src/server/game/Server/Packets/LFGPackets.cpp]: typo ? >> should be << ?

Comment (Zerixx) [inline src/server/game/Server/Packets/LFGPackets.cpp]: You may be right. Not at the computer ATM, will check when I can. I don't remember. It's like 5 am lmao.

Comment (Warpten) [inline src/server/game/Server/Packets/LFGPackets.cpp]: wat

Comment (Zerixx) [inline src/server/game/Server/Packets/LFGPackets.cpp]: No that I'm not completely incoherent, I realize what you mean. I will push an update soon for it. Don't need that method at all.

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 2]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    _worldPacket << MembersCount;
+
+    for (uint32 i = 0; i < JoinSlotsCount; i++)
+        _worldPacket << JoinSlots;
+
+    for (uint32 i = 0; i < MembersCount; i++)
+        _worldPacket << Members;
+
+    _worldPacket.WriteBit(IsBeginning);
+    _worldPacket.WriteBit(ShowRoleCheck);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/LFGPackets.cpp]: flush bits

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 3]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    return &_worldPacket;
+}
+
+WorldPacket const* WorldPackets::LFG::ClientLFGListUpdateStatus::Write()
+{
+    _worldPacket << Ticket;
+    _worldPacket << Request;
+    _worldPacket << Unk;
+    _worldPacket << Reason;
+    _worldPacket.WriteBit(Listed);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/LFGPackets.cpp]: flush

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 4]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    _worldPacket << Slot;
+    _worldPacket << State;
+    _worldPacket << CompletedMask;
+    _worldPacket << PlayersCount;
+
+    for (uint32 i = 0; i < PlayersCount; i++)
+        _worldPacket << Players;
+
+    _worldPacket.WriteBit(ValidCompletedMask);
+    _worldPacket.WriteBit(ProposalSilent);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/LFGPackets.cpp]: flush

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 5]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+{
+    _worldPacket << Ticket;
+    _worldPacket << CountTotalParties;
+    _worldPacket << CountTotalPlayers;
+    _worldPacket << SlotID;
+    _worldPacket << SlotType;
+
+    _worldPacket << Removes;
+
+    _worldPacket.WriteBit(Incremental);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/LFGPackets.cpp]: flush

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 6]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    _worldPacket >> slotsCount;
+
+    for (uint32 i = 0; i < 3; i++)
+        _worldPacket >> Needs[i];
+
+    Comment = _worldPacket.ReadString(commentLength);
+
+    for (uint32 i = 0; i < slotsCount; i++)
+        _worldPacket >> Slots;
+}
```
Comment (MitchesD) [inline src/server/game/Server/Packets/LFGPackets.cpp]: no new line at the end of file

**Comment by MitchesD** [comment]
update these structure is easy, but it's pain to rewrite whole LFG system and implement new needed things, already did some of this, will share something later with you. To sum up I think this is about 1% from the whole patch which you will have to do.

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 7]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    data << playerDungeonInfo.CompletedMask;
+    data << playerDungeonInfo.ShortageRewardCount;
+
+    data << playerDungeonInfo.Rewards;
+
+    for (uint32 i = 0; i < playerDungeonInfo.ShortageRewardCount; i++)
+        data << playerDungeonInfo.ShortageReward;
+
+    data.WriteBit(playerDungeonInfo.FirstReward);
+    data.WriteBit(playerDungeonInfo.ShortageEligible);
```
Comment (Warpten) [inline src/server/game/Server/Packets/LFGPackets.cpp]: missing flush

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 8]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    return data;
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::LFGPlayerRewards const& playerRewards)
+{
+    data << playerRewards.RewardItem;
+    data << playerRewards.RewardItemQuantity;
+    data << playerRewards.BonusCurrency;
+    data.WriteBit(playerRewards.IsCurrency);
+
```
Comment (Warpten) [inline src/server/game/Server/Packets/LFGPackets.cpp]: here too

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 9]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+    data << playerQuestRewards.Mask;
+    data << playerQuestRewards.RewardMoney;
+    data << playerQuestRewards.RewardXP;
+    data << playerQuestRewards.Item;
+    data << playerQuestRewards.Currency;
+    data << playerQuestRewards.BonusCurrency;
+
+    data.WriteBit(playerQuestRewards.UnkBit);
+
+    if (playerQuestRewards.UnkBit)
```
Comment (Warpten) [inline src/server/game/Server/Packets/LFGPackets.cpp]: required flush here too!

File: src/server/game/Server/Packets/LFGPackets.cpp [hunk 10]
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
...
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::ClientLFGProposalUpdatePlayer const& update)
+{
+    data << update.Roles;
+    data.WriteBit(update.Me);
+    data.WriteBit(update.SameParty);
+    data.WriteBit(update.Responded);
+    data.WriteBit(update.Accepted);
+
```
Comment (Warpten) [inline src/server/game/Server/Packets/LFGPackets.cpp]: take a guess

**Comment by Warpten** [comment]
Packet names must not contain the UserClient, Client, etc... prefixes.

**Comment by Zerixx** [comment]
@MitchesD @Warpten I hadn't commit an update just yet, but I took care of the FlushBits already. The only reason I hadn't sent in the fixes is because I'm working on handlers right now, and didn't wanna commit seriously broken and nasty code.

However, anything you've got to help would go a long long way. Warpten, I didn't realize that I can't use those prefixes. Will get those changed asap.

**Comment by spuresz** [comment]
trinitycore when it will be loaded?

**Comment by MitchesD** [comment]
@Zerixx any news? :)

**Comment by Zerixx** [comment]
@MitchesD Yes! I am still working on it, because I have been incredibly busy recently. But, I am happy to say that I'm getting closer and closer. I deleted the old repo, and made a new one, so I will eventually resubmit a new PR!

**Comment by tkrokli** [comment]
@Zerixx : do you think you can update your git setup with the e-mail address of your github account?
It looks like your commits don't link to your account because some ID info is missing in your git setup.

**Comment by Zerixx** [comment]
@tkrokli Yeah that was fixed a while back, and I also had too much garbage in that repo so I just did it over.

**Comment by MitchesD** [comment]
Closing this PR, he created new branch and will open new PR when it's ready

**Comment by Voidx** [comment]
lfg system is pretty broken. This system stop work every 2 or 3 hours.

**Pull request closed without merging**